js/src/hooks/useCreateNoticeForMultipleErrors.js

```diff
@@ -0,0 +1,59 @@
+/**
+ * External dependencies
+ */
+import { __, sprintf } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import useDispatchCoreNotices from '.~/hooks/useDispatchCoreNotices';
+import concatenateListOfWords from '.~/utils/concatenateListOfWords';
+
+/**
+ * Creates an error notice for a list of error messages. For example for a list of error messages: createNoticeForMultipleErrors(['Target Audience', 'Shipping Rates']) will create a notice with the following message:
+ * 'There are errors in the following actions: Target Audience and Shipping Rates. Please try again later.'
+ *
+ * @callback CreateNoticeForMultipleErrors
+ * @param  {Array<string>} errorMessages Error messages to be displayed in the notice.
+ * @param {boolean} [isPartiallySuccessful] Whether some actions were successful.
+ */
+
+/**
+ * @typedef {Object} UseCreateNoticeForMultipleErrors
+ * @property {CreateNoticeForMultipleErrors} createNoticeForMultipleErrors Function to create a notice for a list of errors.
+ */
+
+/**
+ * A hook to create an error notice for multiple messages.
+ *
+ * @return {UseCreateNoticeForMultipleErrors} - A function that will create an error notice for a list of messages.
+ */
+export default function useCreateNoticeForMultipleErrors() {
+	const { createNotice } = useDispatchCoreNotices();
+
+	/**
+	 * @type {CreateNoticeForMultipleErrors} CreateNoticeForMultipleErrors
+	 */
+	const createNoticeForMultipleErrors = (
+		errorMessages,
+		isPartiallySuccessful = true
+	) => {
+		if ( errorMessages.length ) {
+			const listErrors = concatenateListOfWords( errorMessages );
+
+			const content = sprintf(
+				// translators: 1: list of errors 2: optional text if some promises have been invoked successfully.
+				__(
+					'There are errors in the following actions: %1$s. %2$s Please try again later.',
+					'google-listings-and-ads'
+				),
+				listErrors,
+				isPartiallySuccessful ? 'Other changes have been saved.' : ''
+			);
+
+			createNotice( 'error', content );
```

Comment:
❔ 
I'm curious Is there any reason to call `createNotice` in this function? As I understand, `useCreateNoticeForMultipleErrors` and `useCreateNoticeForRejectedPromises` could be non-hook functions because they don't have any states and are not triggered by any React life cycle. Look like they are hook functions just because this hook calls `createNotice` here.

Moving the call to *js/src/edit-free-campaign/index.js* could make these two functions simpler and be used easier as well, and even merge them to single util if there is no other use case that needs to use them separately. Another point is that the implementation of error handling would have the same abstraction level for the `createNotice` calls if put them in the same place.

Reply:
My main reason was to facilitate the consumer to create the notice including the error messages without the need to add the extra step.  Just a hook with two arguments: the promises & the error messages and the notice would display. 

I gave it a try to your suggestion, as I think you have valid points too. Here is the commit: [f83704c](https://github.com/woocommerce/google-listings-and-ads/pull/1602/commits/f83704c62dfcf384f91764a946cc07999438f8f2)